File: app/src/test/java/com/tekutekunikki/mizunomi/VoiceIntakeParserTest.java
package com.tekutekunikki.mizunomi;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

import org.junit.Test;

public class VoiceIntakeParserTest {
    @Test
    public void parsesWaterAmount() {
        VoiceIntakeCandidate candidate = MainActivityKt.parseVoiceIntake("\u6C34300");

        assertEquals("\u6C34", candidate.getDrinkType());
        assertEquals(300, candidate.getAmountMl());
    }

    @Test
    public void skipsTimeAndUsesFirstValidAmount() {
        VoiceIntakeCandidate candidate = MainActivityKt.parseVoiceIntake("13\u6642\u306B300\u98F2\u3093\u3060");

        assertEquals(300, candidate.getAmountMl());
    }

    @Test
    public void returnsNullWhenAmountIsTooLarge() {
        assertNull(MainActivityKt.parseVoiceIntake("\u6C343000000"));
    }

    @Test
    public void returnsNullWhenAmountIsBelowMinimum() {
        assertNull(MainActivityKt.parseVoiceIntake("\u6C3430"));
    }
}

File: app/src/test/java/com/tekutekunikki/mizunomi/domain/VoiceIntakeParserTest.kt
package com.tekutekunikki.mizunomi.domain

import org.junit.Test
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull

class VoiceIntakeParserTest {
    @Test
    fun parsesSupportedVoiceInputs() {
        parseVoiceIntake("水300")!!.also {
            assertEquals(DrinkTypeWater, it.drinkType)
            assertEquals(300, it.amountMl)
        }
        parseVoiceIntake("ポカリ500")!!.also {
            assertEquals(DrinkTypeSportsDrink, it.drinkType)
            assertEquals(500, it.amountMl)
        }
        parseVoiceIntake("コーヒー牛乳200")!!.also {
            assertEquals(DrinkTypeMilkDrink, it.drinkType)
            assertEquals(200, it.amountMl)
        }
        parseVoiceIntake("お茶５００")!!.also {
            assertEquals(DrinkTypeTea, it.drinkType)
            assertEquals(500, it.amountMl)
        }
    }

    @Test
    fun skipsTimeAndUsesFirstValidAmount() {
        assertEquals(300, parseVoiceIntake("13時に300飲んだ")!!.amountMl)
    }

    @Test
    fun returnsNullForMissingOrOutOfRangeAmount() {
        assertNull(parseVoiceIntake("水を飲んだ"))
        assertNull(parseVoiceIntake("水30"))
        assertNull(parseVoiceIntake("水3000000"))
    }
}
